src/Input/AbstractParamAwareInput.php

```diff
@@ -167,4 +169,110 @@ public function getStream()
+    {
+        $mode = $param->getOptionMode();
+
+        if ($mode & InputOption::VALUE_IS_ARRAY) {
+            return ! in_array($value, [null, []], true);
+        }
+
+        return $value !== null;
+    }
+
+    /**
+     * @param mixed $value
+     * @return mixed
+     */
+    private function normalizeValue($value, bool $valueIsArray, ?callable $normalizer)
+    {
+        // No normalizer: nothing to do
+        if ($normalizer === null) {
+            return $value;
+        }
+
+        // Non-array value: normalize it directly
+        if (! $valueIsArray && ! is_array($value)) {
+            return $normalizer($value);
+        }
+
+        // Array value: map each to the normalizer
+        return array_map($normalizer, $value);
+    }
+
+    /**
+     * @param mixed $value
+     * @throws InvalidArgumentException When an array value is expected, but not
+     *     provided.
+     */
+    private function validateValue(
+        $value,
+        bool $valueIsArray,
+        ?callable $validator,
+        string $paramName
+    ): void {
+        // No validator: nothing to do
+        if (! $validator) {
+            return;
+        }
+
+        // Non-array value; validate it directly
+        if (! $valueIsArray) {
+            $validator($value);
+            return;
+        }
+
+        // Array value expected, but not an array: raise an exception
+        if (! is_array($value)) {
+            throw new InvalidArgumentException(sprintf(
+                'Option --%s expects an array of values, but received "%s";'
+                . ' check to ensure the command has provided a valid default.',
+                $paramName,
+                get_debug_type($value)
+            ));
+        }
+
+        // Array value: validate each item in the array
+        array_walk($value, $validator);
+    }
+
+    /**
+     * @return mixed Returns result of asking question, or, if this is a
+     *     multi-select, it loops until no more answers are provided, and retuns
+     *     an array of results.
+     */
+    private function askQuestion(Question $question, bool $valueIsArray, bool $valueIsRequired)
+    {
+        if (! $valueIsArray) {
+            return $this->helper->ask($this, $this->output, $question);
+        }
+
+        $validator = $question->getValidator();
+        $value     = null;
+        $values    = [];
+        do {
+            if (null !== $value) {
+                $values[] = $value;
+            }
+            $value = $this->helper->ask($this, $this->output, $question);
+
+            if ($valueIsRequired && [] === $values) {
+                $question->setValidator(function ($value) use ($validator) {
+                    if (null === $value || '' === $value) {
+                        return $value;
+                    }
+
+                    return $validator($value);
+                });
+            }
+        } while (! in_array($value, [null, ''], true));
```

Comment:
This is the "magic" that allows multi-prompting. Essentially, we prompt until we do not get a value. However, to ensure that we don't attempt to validate an empty line, I need to check for that after the first submission, so I prepend a validator for empty lines to the existing validator (but only  after the first iteration through the loop). Afterwards, I reset the validator.

In this way, we'll keep prompting until the user stops submitting values.